compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -642,6 +642,32 @@ static void PrintBindServiceMethod(const ServiceDescriptor* service,
   p->Print("}\n");
 }
 
+static void PrintAbstractServiceClass(const ServiceDescriptor* service,
+                                   map<string, string>* vars,
+                                   Printer* p,
+                                   ProtoFlavor flavor) {
+  if (flavor != ProtoFlavor::NORMAL) {
```

Comment:
Why limit it to NORMAL? I know it seems stranger for NANO and LITE, but I think we are generating server-side components for them already, if for no other reason for testing. I could agree it would increase the likelihood of accidentally preventing server-side from ProGuarding away.